pkg/marathon/client.go

```diff
@@ -43,6 +44,81 @@ func NewClient(ctx api.Context) (*Client, error) {
 	return &Client{API: client, baseURL: baseURL}, nil
 }
 
+func fetchAppDefinition(url string) ([]byte, error) {
+	resp, err := http.Get(url) // nolint: gosec // using the user-provided URL here is exactly what we want
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("Could not fetch app definition from %s", url)
+	}
+	return ioutil.ReadAll(resp.Body)
+}
+
+// getAppDefinition loads the app definition JSON from either a file pointed to by location or an HTTP
+// URL pointed to by location or, if location is empty, from stdin.
+func getAppDefinition(ctx api.Context, location string) ([]byte, error) {
+	var appBytes []byte
+	var err error
+
+	if location == "" {
+		appBytes, err = ioutil.ReadAll(ctx.Input())
+		if err != nil {
+			return nil, err
+		}
+		return appBytes, nil
+	}
+
+	// get definition from file and fall back to HTTP
+	appBytes, err = ioutil.ReadFile(location)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return nil, err
+		}
+		appBytes, err = fetchAppDefinition(location)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return appBytes, nil
+}
+
+// AddApp creates a deployment from the app definition referenced in appFile which can be a local
+// file name or an HTTP URL. If appFile is empty, the definition is read from ctx.Input().
+func (c *Client) AddApp(ctx api.Context, appLocation string) error {
+	appBytes, err := getAppDefinition(ctx, appLocation)
+	if err != nil {
+		return err
+	}
+	var app marathon.Application
+	err = json.Unmarshal(appBytes, &app)
+	if err != nil {
+		return err
+	}
+
+	existingApp, err := c.API.ApplicationBy(app.ID, &marathon.GetAppOpts{})
+	if err != nil {
+		if apiErr, ok := err.(*marathon.APIError); ok {
+			if apiErr.ErrCode != marathon.ErrCodeNotFound {
+				return err
+			}
+		} else {
+			return err
+		}
+	}
+	if existingApp != nil {
+		return fmt.Errorf("application '%s' already exists", app.ID)
+	}
+
+	createdApp, err := c.API.CreateApplication(&app)
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(ctx.Out(), "Created deployment %s\n", createdApp.Deployments[0]["id"])
```

Comment:
I think the client should not interact with user. This should be printed in command.
```suggestion
```